### PR TITLE
Clarifying database tables for guidewindows and guidestar

### LINKED ISSUE #244: Add GuideWindow to destination field
Add GuideWindow to the archive_catalog destination field for attributes that will appear in the GuideWindow files.  At the same time, remove ScienceCommon from the destination field for fields that will only go to GuideWindow (I think this is just a subset of the fields in guidewindow.yaml, but I wouldn't swear to it).

### PR DESCRIPTION
**Pull request by PaulHuwe**
Resolves [RAD-111](https://jira.stsci.edu/browse/RAD-111)

Closes #244 

This PR clarifies database tables for guidewindows and guidestar.

**Checklist**
- [x] Schema changes discussed at RAD Review Board meeting
- [x] Added entry in `CHANGES.rst` under the corresponding subsection
- [ ] Updated relevant roman_datamodels utilities and tests
- [x] Passed romancal regression testing on Jenkins / PLWishMaster. Link: https://plwishmaster.stsci.edu:8081/job/RT/job/romancal/176/

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -11,6 +11,8 @@
 
 - Changelog CI workflow has been added. [#240]
 
+- Clarifying database tables for guidewindows and guidestar." [#250]
```
Comment (PaulHuwe) [inline CHANGES.rst:14]:
```suggestion
- Clarifying database tables for guidewindows and guidestar. [#250]
```

**Review by ddavis-stsci — APPROVED** [review]
LGTM

**Comment by kdupriestsci** [comment]
This doesn't close #244 since it only touches guidestar.yaml and guidewindow.yaml.  The GuideWindow destination needs to be added to all yaml files that will be included in a GuideWindow file, i.e. everything in common.yaml.

**Comment by kdupriestsci** [comment]
The name of the new table is GuideWindow, not GuideWindowCommon.

**Comment by PaulHuwe** [comment]
Gotcha, will update.

**Review by kdupriestsci — APPROVED** [review]
Looks good.

**Comment by WilliamJamieson** [comment]
This pull request has broken `roman_datamodels`, see https://github.com/spacetelescope/roman_datamodels/actions/runs/4942753423/jobs/8836573621 for more information. Some tests were specifically relying on consistency of this data.

**Comment by PaulHuwe** [comment]
Yes, I noted this above and will be addressing it.

**Pull request merged**